bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/manager/ShellyManagerPage.java

```diff
@@ -413,7 +413,7 @@ protected FwRepoEntry getFirmwareRepoEntry(String deviceType, String mode) throw
             fw = fromJson(gson, entry, FwRepoEntry.class);
 
             // Special case: RGW2 has a split firmware - xxx-white.zip vs. xxx-color.zip
-            if (!mode.isEmpty() && deviceType.equalsIgnoreCase(SHELLYDT_RGBW2)) {
+            if (deviceType.equalsIgnoreCase(SHELLYDT_RGBW2) && !mode.isEmpty()) {
```

Comment:
Could be hardened some more even:
```suggestion
            if (SHELLYDT_RGBW2.equalsIgnoreCase(deviceType) && !mode.isEmpty()) {
```